subprojects/smoke-test/smoke-test.gradle

```diff
@@ -47,6 +47,7 @@ processSmokeTestResources {
 }
 
 tasks.create('smokeTest', SmokeTest) { SmokeTest task ->
+    dependsOn(':cleanUpCaches')
```

Comment:
❌ I don't understand. Smoke test already depends on `cleanUpCaches`: https://e.grdev.net/s/vwurnlajkfdmc/timeline?task=ettkfwljp254y&taskFilter=caches
Since `SmokeTest` is a `DistributionTest`, the distribution testing plugin takes care of that: https://github.com/gradle/gradle/blob/838d9020124dcce40dbf72ea7e7aca2ef51cbb99/buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTestingPlugin.kt#L45-L48

Reply:
Nice cache, I also discovered this just now and I'm investigating it.

Reply:
That's because `removeCachedScripts` method only remove the following directory:

`"scripts", "scripts-remapped", "gradle-kotlin-dsl", "gradle-kotlin-dsl-accessors"`

https://github.com/gradle/gradle/blob/09134dd9b6fc1c80b9c7ab6b9635a8d5ea7695dd/buildSrc/subprojects/cleanup/src/main/kotlin/org/gradle/gradlebuild/testing/integrationtests/cleanup/Cleanup.kt#L68

Now I added a new method to remove `transforms` directory, do you see any risks to do this?

Reply:
Do you know if the `transforms-1` directory is the culprit?
Do we have any `tmp` cleanup on that machines? Like the `tmp` directory being cleaned up periodically?
I am asking since I don't know why we only clean up certain directories in `caches` and not the whole `caches` directory - or even the whole `.gradle-testkit-*` directory.

Reply:
According to https://github.com/gradle/gradle-private/issues/1209#issuecomment-387249708

I'm pretty sure `transforms-1` makes things worse.

Currently we don't have any periodically cleanup of `tmp` (that's why https://github.com/gradle/dev-infrastructure/issues/218 is created). I'm okay with cleaning up the whole `.gradle-testkit-tcagent1` directory in `cleanUpCaches` task, which I think is a cheap and reasonable way to clean  up `tmp` periodically.

Reply:
Yeah, seems like `transforms-1` seems to be the culprit. Let's cleanup that for now.

Reply:
Thanks!